astropy/utils/mixins.py

```diff
@@ -0,0 +1,71 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""A module containing miscellaneous mixin classes."""
+
+from typing import Mapping, Optional
+
+
+__all__ = ["DisplayFormatAttributeMixin"]
+
+
+class DisplayFormatAttributeMixin:
+    """Mixin class for storing format-specific representations of an object.
+
+    Attributes
+    ----------
+    _format : dict[str, str]
+        A mapping to format-specific representations of this object.
+        For example, for the ``Ohm`` unit in :mod:`astropy.units`, it might be
+        nice to have it displayed as ``\\Omega`` by the ``latex`` formatter.
+        In that case, `format` argument should be set to::
+
+            {'latex': r'\\Omega'}
+    name : str
+        The fallback value if a format-specific representation does not exist
+        in the `format` registry.
+
+    Methods
+    -------
+    get_format_name(format='latex')
+        Get a format-specific representation for this object.
+
+    Examples
+    --------
+    ``DisplayFormatAttributeMixin`` is a mixin class, so it must be subclassed.
+
+        >>> class Example(DisplayFormatAttributeMixin):
+        ...     def __init__(self, name, format=None):
+        ...         self.name = name
+        ...         self._format = format or {}  # always a dict
+
+        >>> ex = Example("example", format=dict(latex=r"ex$^{ample}$"))
+
+    Now we can get the format by name.
+
+        >>> ex.get_format_name("latex")
+        'ex$^{ample}$'
+
+    For non-defined formats, the default is the name of the instance.
+
+        >>> ex.get_format_name("not defined")
+        'example'
+    """
+
+    name: str
+    _format: Mapping[str, str]
```

Comment:
Should this perhaps be a `property` with a `setter` that checks the input and deals with `None`?

Reply:
Hm. definitely could be. But then the setter needs a place to store the formats. Like ``_format_`` or something.

Reply:
This is a form of format specification, as #13050 nicely shows, so maybe a good attribute is ``_formats_specs``

```
@property
def _format(self): return self._formats_specs
```